projects/engine/src/engine/engine.ts

```diff
@@ -14,9 +14,9 @@ export class Engine {
   constructor(private manager: BasePluginManager) {
     this.plugins['manager'] = manager
     // Activate the Engine & start listening on activation and deactivation
-    this.activatePlugin('manager').then(() => {
-      this.manager['engineActivatePlugin'] = (name: string) => this.activatePlugin(name)
-      this.manager['engineDeactivatePlugin'] = (name: string) => this.deactivatePlugin(name)
+    this.manager['engineActivatePlugin'] = (name: string) => this.activatePlugin(name)
```

Comment:
is this change needed or is it for cleaning the code?

Reply:
It's needed

Reply:
could you describe why? I see that you are defining the callback before activation but how does that fix the issue (and need a change in the test)?

Reply:
I was doing this.activatePlugin('manager'). Now I'm doing manager.activatePlugin('manager') which does more stuff (including running engine.activatePlugin('manager'))